### ISSUE TITLE
[HOLD for payment 2025-02-07] [$250] Implement animation to the switch expandable sections

### ISSUE THREAD

**Issue by mountiny** [Expensify/App#53759]
This is a request from @dubielzyk-expensify to add animation to the expanding section in Workflows (and other similar places in the app) where after toggling the switch, we open the section like this https://github.com/user-attachments/assets/43e48c18-6b18-4ec6-bc7f-3238f125d616

cc @sumo-slonik @blazejkustra would like to implement this - can you please comment

@dubielzyk-expensify do you have some specifics about the easing/ animation at this moment?
<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~021866452926963427651</li>
        <li>Upwork Job ID: 1866452926963427651</li>
        <li>Last Price Increase: 2024-12-10</li>
    </ul>
</details>

<details><summary>Issue Owner</summary>Current Issue Owner: @stephanieelliott</details>

**Comment by sumo-slonik** [comment]
Hi, I am very eager to get started on this :)
I have an idea how to do it using reanimated and I'm already on it

**Comment by mountiny** [comment]
I think we can just use the autoassigner, its a new task and no rush

**Comment by mountiny** [comment]
@sumo-slonik we also use it in  accounting settings

**Comment by mountiny** [comment]
you have to connected to QBO for example

**Comment by sumo-slonik** [comment]
It is already working everywhere the last place with a problem is the accounting tab, there I have a problem when importing because the component changes before the widget collapses and this causes the illusion of not smooth animation, I have to change the logic of state management in this place, but we are getting closer :D

https://github.com/user-attachments/assets/7002a31e-5547-4376-8b52-696c1aad4ff7

**Comment by sumo-slonik** [comment]
while working on animations in imoprt I found such a bug on main branch, I wanted to fix it within this task but it is more work than what is related to animations so I think it deserves a separate issiue:


https://github.com/user-attachments/assets/7a78fbec-ca59-4ffa-9b7f-747fdc88d029

**Comment by sumo-slonik** [comment]
All switch animations completed:


https://github.com/user-attachments/assets/01a96de6-9e03-4e23-994f-c6cbcd195e0e

**Comment by sumo-slonik** [comment]
@mountiny What do we want to do about this bug?
https://github.com/Expensify/App/issues/53759#issuecomment-2538865532

**Comment by mountiny** [comment]
@sumo-slonik I dont think we ahve to worry about this, its a replay effect issue no need to spend time on it

**Comment by mountiny** [comment]
@dubielzyk-expensify how does the animation look to you?

**Comment by sumo-slonik** [comment]
Also draft PR already has recordings from all platforms:
https://github.com/Expensify/App/pull/53938

**Comment by sumo-slonik** [comment]
PR is ready:
https://github.com/Expensify/App/pull/53938

**Comment by blazejkustra** [comment]
@thesahindia I think you can have a look at the code and start reviewing 😄 Let us know if you have any questions

**Comment by stephanieelliott** [comment]
PR is still under review!

**Comment by stephanieelliott** [comment]
PR is still being actively reviewed

**Comment by dubielzyk-expensify** [comment]
Commented on the PR 👍

**Comment by sumo-slonik** [comment]
We are continuing to wait here for the tick under the new PR after the reverting of the previous one, so we should not mark it as ready yet.

**Comment by stephanieelliott** [comment]
Updating labels to remove payment status, seeing as we're still waiting on the second PR: https://github.com/Expensify/App/pull/55412

**Comment by mountiny** [comment]
The blockers were handled

**Comment by stephanieelliott** [comment]
Summarizing payment on this issue: 

 - Contributor: @sumo-slonik: EC, no payment due 
 - Contributor+: @thesahindia $250 via ND -- please request!

**Comment by mountiny** [comment]
I think the compensation for this would be $250

**Comment by garrettmknight** [comment]
$250 approved for @thesahindia
